lnrpc/invoicesrpc/invoices.proto

```diff
@@ -35,6 +35,12 @@ service Invoices {
     settled, this call will succeed.
     */
     rpc SettleInvoice (SettleInvoiceMsg) returns (SettleInvoiceResp);
+
+    /*
+    LookupInvoice attempts to look up at invoice. An invoice can be refrenced
+    using either its payment hash, payment address, or set ID.
+    */
+    rpc LookupInvoice (LookupInvoiceMsg) returns (lnrpc.Invoice);
```

Comment:
I am wondering how users would list all recurring payments to a single main invoice. Probably a filter on `ListInvoices`?

Reply:
So if they want all the payments to a single invoice, they should just be able to use the `payment_addr` here. One change I'm considering tacking on is that optionally if you just pass in the `set_id`, then you get _just_ the HTLC set associated with that `set_id`.

Reply:
Ah right, all AMP payments are contained within a single invoice object. Is that a good idea from an API perspective? Maybe it is possible to add an extra level just on the rpc interface so that the caller can iterate over a list of AMP payments with each payment containing only the htlcs for that payment. A simple un-flatten during marshalling.

Reply:
> Ah right, all AMP payments are contained within a single invoice object. Is that a good idea from an API perspective? 

Yeah this came up in my mind today making further changes. Even with the invoice stored distinctly as is, the current `UpdateInvoice` logic will still end up reading _all_ the HTLCs, since it needs to do things like re-accumulate the amount stored in `AmtPaid`. Instead of doing this, if we have a given `setID`, then we can pass that into `fetchInvoice`, and use a cursor to have it _only_ populate the relevant HTLCs. Given we'll no longer cancel back all active HTLCs with a different `setID` once one is settled, I think this should work np. Note that this is just on the DB level, so a transaction doesn't need to pull in every single HTLC ever settled. 

It also gives callers a way to fetch an invoice but _only_ have the relevant setID HTLCs populated (which could be exposed over the RPC interface). 

Unrolling all the payments is def also an option, though I'm trying to keep the RPC changes to a minimal set in this PR. Goal is that users don't _really_ need to update their logic yet, and gives us a bit more time to come up with a more encompassing interface. Lots of existing baggage like the lack of a single "update" counter, the inability to obtain a time series of accepted invoices, etc, etc.

Reply:
> Related to what I mentioned above, we can then have LookupInvoice let users filter out for just those HTLCs.

I think the missing like here, would then be a new RPC call to list all the relevant setIDs for a given `payment_addr`....

Devs could use that to get all the recurring payment identifiers w/o reading the entire invoice state, then look up the details (including the attached custom records) of all the HTLCs associated with them. 

> Maybe it is possible to add an extra level just on the rpc interface so that the caller can iterate over a list of AMP payments with each payment containing only the htlcs for that payment. A simple un-flatten during marshalling.

My idea for the subscribe invoices backlog delivery is something like this: each time we send the AMP invoice across, it _only_ includes the HTLCs for that particular setID. Or did you have something else in mind? I think that extra call I described above would let them iterate over the full invoice details for each `setID`. We may also want to add some additional metadata like maybe the settle time to allow them to filter down further.